cw-dex/tests/osmosis_proptests.rs

```diff
@@ -0,0 +1,177 @@
+        .into_iter()
+        .zip(initial_liquidity.into_iter())
+        .map(|(a, b)| a.min(b) - 1u64)
+        .collect();
+
+    let wasm = Wasm::new(&runner);
+    let assets: Vec<Coin> = added_liquidity
+        .into_iter()
+        .enumerate()
+        .map(|(i, amount)| Coin {
+            denom: format!("denom{}", i),
+            amount: amount.into(),
+        })
+        .collect();
+
+    let provide_msg = ExecuteMsg::ProvideLiquidity {
+        assets: assets.clone().into(),
+        min_out: Uint128::one(),
+    };
+    wasm.execute(&contract_addr, &provide_msg, &assets, &accs[0])
+        .unwrap();
+
+    // Query LP token balance
+    let lp_token_balance =
+        bank_balance_query(&runner, contract_addr, format!("gamm/pool/{}", pool_id)).unwrap();
+
+    assert_ne!(lp_token_balance, Uint128::zero());
+}
+
+fn test_multi_pool_swap(
+    pool_type: OsmosisPoolType,
+    pool_liquidity: Vec<u64>,
+    offer: Asset,
+    ask: AssetInfo,
+) {
+    let (runner, accs, _pool_id, contract_addr) =
+        setup_pool_and_contract(pool_type, pool_liquidity).unwrap();
+
+    let wasm = Wasm::new(&runner);
+    let funds = vec![offer.clone().try_into().unwrap()];
+
+    let swap_msg = ExecuteMsg::Swap {
+        offer: offer.clone(),
+        ask: ask.clone(),
+        min_out: Uint128::one(),
+    };
+    wasm.execute(&contract_addr, &swap_msg, &funds, &accs[0])
+        .unwrap();
+
+    // Query LP token balance
+    let offer_balance =
+        bank_balance_query(&runner, contract_addr.clone(), offer.info.to_string()).unwrap();
+    let ask_balance = bank_balance_query(&runner, contract_addr, ask.to_string()).unwrap();
+
+    assert_eq!(offer_balance, Uint128::zero());
+    assert_ne!(ask_balance, Uint128::zero());
+}
+
+fn setup_swap_test_params(
+    pool_liquidity: Vec<u64>,
+    offer_idx: usize,
+    offer_amount: u64,
+    ask_idx: usize,
+    n: usize,
+) -> (Vec<u64>, Asset, AssetInfo) {
+    // Indices mod n
+    let offer_idx = offer_idx % n;
+    let ask_idx = ask_idx % n;
+
+    // If offer and ask are the same, increment ask mod n
+    let ask_idx = if offer_idx == ask_idx {
+        (ask_idx + 1) % n
+    } else {
+        ask_idx
+    };
+
+    // Offer amount cannot be larger than the pool liquidity
+    let offer_amount = offer_amount.min(pool_liquidity[offer_idx]);
+
+    // Increment any 0s in the pool liquidity
+    let pool_liquidity: Vec<u64> = pool_liquidity
+        .into_iter()
+        .map(|l| if l == 0 { 1 } else { l })
+        .collect();
+
+    // Construct offer and ask
+    let ask = AssetInfo::Native(format!("denom{}", ask_idx));
+    let offer = Asset {
+        info: AssetInfo::Native(format!("denom{}", offer_idx)),
+        amount: Uint128::from(offer_amount),
+    };
+
+    (pool_liquidity, offer, ask)
+}
+
+proptest! {
+    #![proptest_config(ProptestConfig {
+        // Setting both fork and timeout is redundant since timeout implies
+        // fork, but both are shown for clarity.
```

Comment:
Outdated comment? Don't see mention of "timeout" nor "fork" anywhere.

Reply:
ack